assets/js/pages/Users/UserForm.test.jsx

```diff
@@ -225,6 +230,82 @@ describe('UserForm', () => {
     });
   });
 
+  it('should allow to disable TOTP correctly', async () => {
+    const user = userEvent.setup();
+    const mockOnSave = jest.fn();
+
+    const {
+      fullname,
+      email,
+      username,
+      created_at: createdAt,
+      updated_at: updatedAt,
+      totp_enabled_at: totpEnabledAt,
+    } = userFactory.build();
+
+    await act(async () => {
+      render(
+        <UserForm
+          saveText="Save"
+          fullName={fullname}
+          emailAddress={email}
+          username={username}
+          createdAt={createdAt}
+          updatedAt={updatedAt}
+          totpEnabledAt={totpEnabledAt}
+          editing
+          onSave={mockOnSave}
+        />
+      );
+    });
+    await user.click(screen.getAllByText('Enabled')[1]);
+    await user.click(screen.getByText('Disabled'));
+
+    await user.click(screen.getByRole('button', { name: 'Save' }));
+
+    expect(mockOnSave).toHaveBeenNthCalledWith(1, {
+      fullname,
+      email,
+      enabled: true,
+      abilities: [],
+      totp_enabled: false,
+    });
+  });
+
+  it('should not allow to enable TOTP from user mgmt', async () => {
+    const user = userEvent.setup();
+    const mockOnSave = jest.fn();
+
+    const {
+      fullname,
+      email,
+      username,
+      created_at: createdAt,
+      updated_at: updatedAt,
+      totp_enabled_at: totpEnabledAt,
+    } = userFactory.build({ totp_enabled_at: null });
+
+    await act(async () => {
+      render(
+        <UserForm
+          fullName={fullname}
+          emailAddress={email}
+          username={username}
+          createdAt={createdAt}
+          updatedAt={updatedAt}
+          totpEnabledAt={totpEnabledAt}
+          editing
+        />
+      );
+    });
+
+    await user.click(screen.getByText('Disabled'));
+
+    // This should not change the selected option as it should be disabled
+    await user.click(screen.getAllByText('Enabled')[1]);
```

Comment:
I think `expect(screen.getAllByText('Enabled')[1]).toBeDisabled()` is more appropriate